wrappers/obj-c/ODWLogger.mm

```diff
@@ -31,6 +34,12 @@ -(instancetype)initWithILogger:(ILogger*)logger
 	        NSLog(@"Logger initialized successfully");
 		}
         _semanticContext = [[ODWSemanticContext alloc] initWithISemanticContext:_wrappedLogger->GetSemanticContext()];
+
+        [[NSNotificationCenter defaultCenter] addObserver:self
+                                                 selector:@selector(applicationWillTerminate:)
```

Comment:
Since the headers don't agree between iOS and Mac, what about having a callback here that is called when the host application identifies the `applicationWillTerminate` scenario? Then, in the future as the platforms converge/diverge, the SDK won't need to have any changes and the host application can work out the details of know the scenario and update accordingly.

Reply:
Had thought about this approach to have a static / class method which the app can call when it encounters the appWillTerminate scenario.
I an offline discussion @eduardo-camacho suggests to keep as is. Thoughts?

Reply:
hmm, lemme think a bit. My overall intent is to avoid the need for any platform specific log here. I understand that there may be instance where we must do some platform specific implementation, and in that case, I would recommend covering both iOS and Mac.

Reply:
Isn't there `#ifdef` you can use to make it iOS only? I think though that force termination applies to Mac as well, i.e. you can `Force Quit` if you press that shortcut (Option while right-clicking on icon). Press these three keys together: Option, Command, and Esc (Escape). Or choose Force Quit from the Apple menu  in the upper-left corner of your screen. (This is similar to pressing Control-Alt-Delete on a PC.) Then select the app in the Force Quit window and click Force Quit...

I don't know though if it triggers the same UX notification or not.

Reply:
One more disadvantage of listening  to notification in the `ODWLogger` is that not every host app may need or want this behaviour. Since, we are adding the check in `PerformActionWithCppExceptionsCatch` function  which is kind of a bottle-neck to access the SDK in anyway.  Other apps may want to access the SDK for some other purposes in appWillTerminate notification and the current check will make all such accesses no-op, giving a false sense to the host app.

Reply:
@sid-dahiya , @maxgolov created an API in `ODWLogManager` instead of listening to the notification in the SDK. The host app can call the API on `appWillTerminate` notification.